### PR TITLE
I added two algorithm related to matrix.

### PR DESCRIPTION
**Pull request by sadiul-hakim**
### **Describe your change:**

- [x ] Add an algorithm?

### **Checklist:**

- [ x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [ x] This pull request is all my own work -- I have not plagiarized.
- [ x] I know that pull requests will not be merged if they fail the automated tests.
- [ x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x ] All new Java files are placed inside an existing directory.
- [x ] All functions and variable names follow Java naming conventions.

### CODE REVIEW

**Review by siriak — CHANGES_REQUESTED** [review]
The code looks good, could you please add some JUnit tests? (see https://github.com/TheAlgorithms/Java/tree/master/src/test/java/com/thealgorithms)

**Comment by sadiul-hakim** [comment]
Thank you for your reply. I'll try to add some test code as soon as possible..

File: src/main/java/com/thealgorithms/searches/SearchInARowAndColWiseSortedMatrix.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+                { 8, 9, 10, 11, 12 },
+                { 14, 15, 16, 17, 18 },
+                { 23, 24, 25, 26, 27 },
+                { 30, 31, 32, 33, 34 }
+        };
+
+        var search = new SearchInARowAndColWiseSortedMatrix();
+        int[] res = search.search(matrix, 26);
+        System.out.println(Arrays.toString(res));
+    }
```
Comment (siriak) [inline src/main/java/com/thealgorithms/searches/SearchInARowAndColWiseSortedMatrix.java:53]:
```suggestion

```

File: src/test/java/com/thealgorithms/searches/TestSearchInARowAndColWiseSortedMatrix.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+                { 23, 24, 25, 26, 27 },
+                { 30, 31, 32, 33, 34 }
+        };
+
+        var test = new SearchInARowAndColWiseSortedMatrix();
+        int[] res = test.search(matrix, 96);
+        int[] expectedResult = { -1, -1 };
+        assertArrayEquals(expectedResult, res);
+    }
+}
```
Comment (siriak) [inline src/test/java/com/thealgorithms/searches/TestSearchInARowAndColWiseSortedMatrix.java:38]:
```suggestion
}

```

File: src/main/java/com/thealgorithms/searches/SearchInARowAndColWiseSortedMatrix.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+            if (matrix[i][j] == value) {
+                result[0] = i;
+                result[1] = j;
+                return result;
+            }
+            if (value > matrix[i][j]) {
+
+                i++;
+
```
Comment (siriak) [inline src/main/java/com/thealgorithms/searches/SearchInARowAndColWiseSortedMatrix.java:32]:
```suggestion
                i++;
```

File: src/main/java/com/thealgorithms/searches/SearchInARowAndColWiseSortedMatrix.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    public int[] search(int[][] matrix, int value) {
+        int n = matrix.length;
+        // This variable iterates over rows
+        int i = 0;
+        // This variable iterates over columns
+        int j = n - 1;
+        int[] result = { -1, -1 };
+
+        while (i < n && j >= 0) {
+
```
Comment (siriak) [inline src/main/java/com/thealgorithms/searches/SearchInARowAndColWiseSortedMatrix.java:23]:
```suggestion
```

**Review by siriak — CHANGES_REQUESTED** [review]
The search and its tests looks good. Could you make PrintAMatrixInSpiralOrder return what it intends to print as an array and add tests for that? Or just remove it and I can merge the search. Thanks!

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Pull request merged**